core/src/main/java/org/sonarsource/sonarlint/core/analyzer/sensor/ScannerExtensionDictionnary.java

```diff
@@ -196,41 +184,9 @@ private static void evaluateClass(Class extensionClass, Class annotationClass, L
     }
   }
 
-  private void evaluateMethod(Object extension, Method method, List<Object> results) {
-    try {
-      Object result = method.invoke(extension);
-      if (result == null) {
-        return;
-      }
-      if (result instanceof Class<?>) {
-        results.addAll(componentContainer.getComponentsByType((Class<?>) result));
-      } else if (result instanceof Collection<?>) {
-        results.addAll((Collection<?>) result);
-      } else if (result.getClass().isArray()) {
-        for (int i = 0; i < Array.getLength(result); i++) {
-          results.add(Array.get(result, i));
-        }
-      } else {
-        results.add(result);
-      }
-    } catch (Exception e) {
-      throw new IllegalStateException("Can not invoke method " + method, e);
-    }
-  }
-
-  private static void checkAnnotatedMethod(Method method) {
-    if (!Modifier.isPublic(method.getModifiers())) {
-      throw new IllegalStateException("Annotated method must be public:" + method);
-    }
-    if (method.getParameterTypes().length > 0) {
-      throw new IllegalStateException("Annotated method must not have parameters:" + method);
-    }
-  }
-
-  private static boolean shouldKeep(Class type, Object extension, @Nullable Project project, @Nullable ExtensionMatcher matcher) {
+  private static boolean shouldKeep(Class type, Object extension, @Nullable Project project) {
     boolean keep = (ClassUtils.isAssignable(extension.getClass(), type)
-      || (org.sonar.api.batch.Sensor.class.equals(type) && ClassUtils.isAssignable(extension.getClass(), Sensor.class)))
-      && (matcher == null || matcher.accept(extension));
+      || (org.sonar.api.batch.Sensor.class.equals(type) && ClassUtils.isAssignable(extension.getClass(), Sensor.class)));
     if (keep && project != null && ClassUtils.isAssignable(extension.getClass(), CheckProject.class)) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this use of "CheckProject"; it is deprecated. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)